### DISCUSSION TITLE
[How to] use Blanket in Android device with Termux

### DISCUSSION THREAD

**Discussion by archisman-panigrahi** [rafaelmardojai/blanket#114] [General]
There is a way to use Blanket in almost any Android device.

**Summary:**
- Install Termux in Android
- Install rootfs for your favorite distro (I tried Ubuntu 20.04, but others should work too).
- Install a DE
- Set up VNC Server
- Install Blanket
- Set up sound

**Demonstration** - **[Youtube video](https://www.youtube.com/watch?v=8T9x14EIeS0)**

Details
------------------------------------------------

1. Install [Termux](https://termux.com/) from F-Droid or Play Store (Google Play version of Termux is not anymore updated).
2. Install Ubuntu 20.04 with proot in Termux. I used [Andronix](https://andronix.app/) to automate the process.
3. Now install a desktop environment (XFCE worked for me), and set up a VNC server (Andronix can automate these tasks for you). To start a new XFCE session, enter the command `vncserver-start`.
4. Install a VNC viewer app in Android, e.g. [androidVNC](https://f-droid.org/en/packages/android.androidVNC/), or [VNC viewer](https://play.google.com/store/apps/details?id=com.realvnc.viewer.android&hl=en_US&gl=US).
5. Install Blanket (I used the stable PPA in Ubuntu)
```
add-apt-repository ppa:apandada1/blanket
apt update
apt install blanket
```
6. Exit Ubuntu, and [set up sound](https://docs.andronix.app/hardware/sound) in `Termux shell` (not inside Ubuntu)
```
 pkg install wget && wget https://andronixos.sfo2.cdn.digitaloceanspaces.com/OS-Files/setup-audio.sh && chmod +x setup-audio.sh && ./setup-audio.sh
 ```

To run Blanket after installation
--------------------------------------
Once installation is done,
- Open Termux
- Start Ubuntu
- Enter the command `pulseaudio --start` in a new Termux session (not in the Ubuntu session)
- Start VNC Server
- Open VNC viewer app in Android, and open Blanket
- When closing, turn off the vncserver (`vncserver-stop`) in Ubuntu shell, and then close ubuntu (`exit` in Termux shell).
If you forget to close vncserver before closing Ubuntu, you may have to manually remove the lockfile for vncserver, before you can start VNCServer once again.

Additional tips/comments:
-----------------------------------------
- You can set a custom resolution, which suites your screen size. For me, the fonts were too small with the default (`1900x1200`) resolution, and I changed it to `1500x700` in my 6.5 inch phone.
- You can also appropriately set the custom resolution to fit your device in portrait mode (eg. `700x1500`)
- Ubuntu + XFCE consumes ~ **1.2 GB** of storage, and **300 MB** of RAM. It may be less if you use a window manager like i3 instead of XFCE, but I have not tried that.
- You can install your favorite GTK theme.
 
Why Ubuntu?
--------------------------------------
I used an Ubuntu rootfs due to its relatively less download size, and because there I won't have to build from source as the PPA is available.
_**Feel free** to add workarounds for your favorite distro in subsequent comments._

**Comment by rafaelmardojai** [comment]
Really nice!

**Reply by archisman-panigrahi** [reply]
@rafaelmardojai I found a similar FOSS project for Android. It has a sleep timer and a wake up timer.
https://noice.ashutosh.blog/
https://github.com/ashutoshgngwr/noice
https://f-droid.org/en/packages/com.github.ashutoshgngwr.noice/

**Reply by rafaelmardojai** [reply]
@apandada1  Looks like a nice reference to improve Blanket.

**Comment by lamyergeier** [comment]
Please if possible provide it as an app on Android
